src/routes/orderRoutes.js

```diff
@@ -1,8 +1,24 @@
 const express = require('express');
 const router = express.Router();
-const { authenticateUser } = require('../middleware/authentication');
-const { createOrder } = require('../controllers/orderController');
+const {
+  authenticateUser,
+  authorizePermissions,
+} = require('../middleware/authentication');
+const {
+  createOrder,
+  getAllOrders,
+  getSingleOrder,
+  deleteOrder,
+} = require('../controllers/orderController');
 
-router.post('/', authenticateUser, createOrder);
+router
+  .route('/')
+  .post(authenticateUser, createOrder)
+  .get(authenticateUser, authorizePermissions('admin'), getAllOrders);
+
+router
+  .route('/:id')
+  .get(authenticateUser, authorizePermissions('admin'), getSingleOrder)
```

Comment:
For fetching a single order, I think we'll need to allow non-admins to do this too; we can do `checkPermisisons` inside the controller to make sure it's the right user or an admin.

```suggestion
  .get(authenticateUser, getSingleOrder)
```